src/main/java/com/fidesmo/tutorials/counterapp/MainActivity.java

```diff
@@ -151,78 +175,31 @@ private void communicateWithCard(IsoCard isoCard) {
         }
     }
 
-    @UiThread
-    void showInstallButton(boolean visibility) {
-        if (visibility){
-            installButton.setVisibility(View.VISIBLE);
-        } else {
-            installButton.setVisibility(View.GONE);
-        }
-    }
-
     /**
-     * Calls the Fidesmo App in order to install the Counter Applet  into the Fidesmo Card
-     * First, it checks whether the Fidesmo App is installed; if not, it opens Google Play
+     * Reads the value held by the Counter Applet by sending a SELECT operation to the card
      */
     @Click
-    void installButtonClicked() {
-        if (appInstalledOrNot(FIDESMO_APP)) {
-            try {
-                // create Intent to the Action exposed by the Fidesmo App
-                Intent intent = new Intent(SERVICE_DELIVERY_CARD_ACTION, Uri.parse(SERVICE_URI + APPLICATION_ID + "/" + SERVICE_ID));
-                startActivityForResult(intent, SERVICE_DELIVERY_REQUEST_CODE);
-            } catch (IllegalArgumentException e) {
-                Log.e(TAG, "Error when parsing URI");
-            }
-        } else {
-            notifyMustInstall();
-        }
-    }
-
-    // method called when the Fidesmo App activity has finished
-    // Will redraw the screen if it finished successfully
-    @OnActivityResult(SERVICE_DELIVERY_REQUEST_CODE)
-    void onResult(int resultCode) {
-        if (resultCode == RESULT_OK) {
-            Log.i(TAG, "Cardlet installation returned SUCCESS");
-            setMainMessage(R.string.put_card);
-            installButton.setVisibility(View.GONE);
-        } else {
-            Log.i(TAG, "Cardlet installation returned FAILURE");
-            Toast.makeText(getApplicationContext(), getString(R.string.failure),
-                    Toast.LENGTH_LONG).show();
-        }
+    void readCounterButtonClicked() {
+        pendingCommand = READ_COUNTER;
+        // The first argument is the activity for which the NFC is managed
+        // The second argument is the OnDiscoveredTagListener which is also implemented by this activity
+        // This means that tagDiscovered will be called whenever a new tag appears
+        tagDispatcher = TagDispatcher.get(this, this);
+        setMainMessage(getString(R.string.put_card));
+        // Start listening on the NFC interface when the app gains focus.
+        tagDispatcher.enableExclusiveNfc();
```

Comment:
@miguelcardo You should use the new TagArbiter, it will store tags for future use, vastly improving user experience (e.g. by letting the user do several actions on the card without tapping again or allowing the card to be presented before the action is started)

https://github.com/fidesmo/nordpol/blob/master/android/src/main/java/nordpol/TagArbiter.java